text/0038-output-types-for-tokenization-and-sc/0038-output-types-for-tokenization-and-sc.md

```diff
@@ -0,0 +1,2903 @@
+        <td><b>Type</b></td>
+        <td><b>Description</b></td>
+    </tr>
+    <tr>
+        <td>Block Type</td>
+        <td>uint8</td>
+        <td>
+            Set to <strong>value 7</strong> to denote a <i>Metadata Block</i>.
+        </td>
+    </tr>
+    <tr>
+        <td>Data Length</td>
+        <td>uint32</td>
+        <td>
+            Length of the following data field in bytes.
+        </td>
+    </tr>
+    <tr>
+        <td>Data</td>
+        <td>ByteArray</td>
+        <td>Binary data.</td>
+    </tr>
+</table>
+
+#### Indexation Block
+
+An indexation block makes it possible to tag outputs with an index, so they can be retrieved through an API not only by
+their address, but also based on the the `Indexation Tag`. **The combination of an <i>Indexation Block</i>, a
+<i>Metadata Block</i> and a <i>Sender Block</i> makes it possible to retrieve data associated to an address and stored
+in outputs that was created by a specific party (`Sender`) for a specific purpose (`Indexation Tag`).**
+
+Storing indexed data in outputs however incurs greater dust deposit for such outputs, because they create look-up
+entries in nodes' databases.
+
+##### Additional syntactic transaction validation rules:
+- An output with <i>Indexation Block</i> is valid, if and only if 0 < `Indexation Tag Length` ≤
+  `Max Indexation Tag Length`.
+
+<details>
+<summary>Indexation Block</summary>
+<blockquote>
+    Defines an indexation tag to which the <i>Metadata Block</i> will be indexed. Creates an indexation lookup in nodes.
+</blockquote>
+</details>
+
+<table>
+    <tr>
+        <td><b>Name</b></td>
+        <td><b>Type</b></td>
+        <td><b>Description</b></td>
+    </tr>
+    <tr>
+        <td>Block Type</td>
+        <td>uint8</td>
+        <td>
+            Set to <strong>value 8</strong> to denote a <i>Indexation Block</i>.
+        </td>
+    </tr>
+    <tr>
+        <td>Indexation Tag Length</td>
+        <td>uint8</td>
+        <td>
+            Length of the following indexation tag field in bytes.
+        </td>
+    </tr>
+    <tr>
+        <td>Indexation Tag</td>
+        <td>ByteArray</td>
+        <td>Binary indexation tag.</td>
+    </tr>
+</table>
+
+
+
+### Chain Constraint in UTXO
+
+Previously created transaction outputs are destroyed when they are consumed in a subsequent transaction as an input.
+The chain constraint makes it possible to **carry the UTXO state machine state encoded in outputs across transactions.**
+When an output with chain constraint is consumed, that transaction has to create a single subsequent output that
+carries the state forward. The **state can be updated according to the transition rules defined for the given type of
+output and its current state**. As a consequence, each such output has a unique successor, and together they form a path
+or *chain* in the graph induced by the UTXO spends. Each chain is identified by its globally unique identifier.
+
+![](https://i.imgur.com/izQ4DB1.png)
```

Comment:
Would be nice if this image had a white background instead of transparent otherwise it's pretty hard to read on a dark theme (black letters on black background)

Reply:
Updated :)